### PR TITLE
Clear environment before reloading

### PR DESCRIPTION
**Pull request by maiqbal11**
Fixes https://github.com/Azure/azure-functions-python-worker/issues/410

### CODE REVIEW

File: azure/functions_worker/dispatcher.py [hunk 1]
```diff
@@ -356,6 +356,8 @@ def _serialize_exception(self, exc):
             logger.info('Received FunctionEnvironmentReloadRequest, '
                         'request ID: %s', self.request_id)
 
+            os.environ.clear()
+
```
Comment (pragnagopa) [inline azure/functions_worker/dispatcher.py]: Unit tests?

Comment (maiqbal11) [reply azure/functions_worker/dispatcher.py]: Added a UT

File: tests/test_rpc_messages.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import asyncio
+import os
+
+from azure.functions_worker import protos
+from azure.functions_worker import dispatcher
+
+
+class TestGRPC(unittest.TestCase):
+    def test_handle_environment_reload_request(self):
+        test_env = {'TEST_KEY': 'foo', 'HELLO': 'world'}
```
Comment (pragnagopa) [inline tests/test_rpc_messages.py]: Add variations of this test: empty  and null test_env.
